File: Frontend/apps/astrogame_app/lib/models/stellar/base_types/stellar_system.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/stellar/base_types/stellar_object.dart';
import 'package:astrogame_app/models/stellar/base_types/stellar_thing.dart';
import 'package:json_annotation/json_annotation.dart';

@GuidConverter()
abstract class StellarSystem extends StellarThing {
  @JsonKey()
  List<StellarObject> centerObjects;

  @JsonKey()
  List<StellarSystem> satellites;
}

File: Backend/AstroGame.Shared/Models/Stellar/BaseTypes/StellarSystem.cs
using Newtonsoft.Json;
using System.Collections.Generic;

namespace AstroGame.Shared.Models.Stellar.BaseTypes
{
    /// <summary>
    /// Any construct in space that consists of multiple <see cref="StellarThing"/>
    /// </summary>
    public abstract class StellarSystem : StellarThing
    {
        protected StellarSystem()
        {
        }

        [JsonProperty(Order = 9)] public List<StellarObject> CenterObjects { get; set; }

        [JsonProperty(Order = 10)] public List<StellarSystem> Satellites { get; set; }
    }
}